docs/HowToCertificatesRotation.md

```diff
@@ -0,0 +1,486 @@
+```
+
+> **NOTE**: Adjust the values depending on the key pair certificates that have been rotated.
+
+Wait for all database processes to restart, report `Ready` and ensure that the new key pair certificates are used by all domain processes as described in [Verify Domain Certificates](#verify-domain-certificates).
+
+#### Using HashiCorp Vault
+
+HashiCorp uses sidecar containers to inject keys into another container.
+It automatically updates secret values stored in the Vault KV store to match the files in the volume mounted inside the container.
+NuoDB APs will reload the keystore without the need for reconfiguration and restart.
+Therefore, it is enough to update the keystore, client PEM and CA certificate in HashiCorp Vault.
+
+Ensure that the keystore has the same password (as specified by `admin.tlsKeyStore.password` Helm option) so that the APs can reload the keystore without having to be reconfigured and restarted.
+
+For more information on how to configure TLS with HashiCorp Vault, see [Using HashiCorp Vault for Management of TLS Certificates](./HowToHashiCorpVault.md).
+
+Make sure that the new key pair certificates are used by all APs as described in [Verify Domain Certificates](#verify-domain-certificates).
+
+Restart database processes so that the APs generate new engine certificates signed with the new server certificate chain in the [Intermediate CA](./HowtoTLS.md#intermediate-ca) model or the new shared key pair certificates are used in the [Pass-down](./HowtoTLS.md#pass-down) model.
+Kubernetes 1.15+ supports a rollout restart which can be used to restart database processes in a rolling fashion.
+
+```bash
+kubectl rollout restart \
+  --namespace nuodb \
+  statefulset sm-database-nuodb-cluster0-demo-hotcopy
+
+kubectl rollout restart \
+  --namespace nuodb \
+  statefulset sm-database-nuodb-cluster0-demo
+
+kubectl rollout restart \
+  --namespace nuodb \
+  deployment te-database-nuodb-cluster0-demo
+```
+
+Alternatively, the database processes can be restarted manually by shutting down each process using `nuocmd shutdown process --start-id ...` command.
+Kubernetes will automatically restart the engine container which will effectively restart the database process.
+
+It is recommended to perform each rollout or database process restart sequentially after verifying the domain process certificates as described in [Verify Domain Certificates](#verify-domain-certificates).
+
+### Verify Domain Certificates
+
+To get the current certificate data in the NuoDB domain, use `nuocmd get certificate-info` command and make sure that the renewed key pair certificates are displayed in the corresponding sections.
+
+```bash
+kubectl exec -ti admin-nuodb-cluster0-0 \
+  --namespace nuodb -- \
+  nuocmd --show-json get certificate-info
+```
+
+The command displays the certificates information in the below sections:
+
+- `processCertificates` - information about certificates used by each database process identified by `startId`.
```

Comment:
This should say something like `the certificate chain used by each database process` instead of `certificates`, for clarity. Each process has only one certificate for its own identity. Same comment below for `serverCertificates`.

Reply:
Agree.